pandas/_libs/src/headers/portable.h

```diff
@@ -8,6 +8,7 @@
 // GH-23516 - works around locale perf issues
 // from MUSL libc, MIT Licensed - see LICENSES
 #define isdigit_ascii(c) (((unsigned)(c) - '0') < 10u)
+#define getdigit_ascii(c, default) (isdigit_ascii(c) ? ((int)((c) - '0')) : default)
```

Comment:
@vnlitvin since it seems you are more C-comfortable than many contributors, two things that might be up your alley:

- #19361 we have two separate implementations of `xstrtod` that are slightly different.  Last time I tried to de-duplicate them i broke everything.  You might have more luck/talent.
- https://github.com/cython/cython/issues/2570 some of these headers file could/should be implemented upstream in cython

Reply:
@jbrockmendel see #25925, that resolve #19361 problem